x-pack/plugins/security_solution/public/detections/components/rules/step_rule_actions/actions_view.tsx

```diff
@@ -0,0 +1,218 @@
+    s: (
+      <FormattedMessage
+        id="xpack.securitySolution.detectionEngine.ruleDetails.onceInEverySecondsLabel"
+        defaultMessage="Once {secondsCount, plural, one {a} other {in every}} {secondsCount, plural, one {second} other {# seconds}}"
+        values={{ secondsCount: value }}
+      />
+    ),
+    m: (
+      <FormattedMessage
+        id="xpack.securitySolution.detectionEngine.ruleDetails.onceInEveryMinutesLabel"
+        defaultMessage="Once {minutesCount, plural, one {a} other {in every}} {minutesCount, plural, one {minute} other {# minutes}}"
+        values={{ minutesCount: value }}
+      />
+    ),
+    h: (
+      <FormattedMessage
+        id="xpack.securitySolution.detectionEngine.ruleDetails.onceInEveryHoursLabel"
+        defaultMessage="Once {hoursCount, plural, one {an} other {in every}} {hoursCount, plural, one {hour} other {# hours}}"
+        values={{ hoursCount: value }}
+      />
+    ),
+    d: (
+      <FormattedMessage
+        id="xpack.securitySolution.detectionEngine.ruleDetails.onceInEveryDaysLabel"
+        defaultMessage="Once {daysCount, plural, one {a} other {in every}} {daysCount, plural, one {day} other {# days}}"
+        values={{ daysCount: value }}
+      />
+    ),
+  };
+
+  return <DescriptionLine>{messagesByUnit[unit] || i18n.PERIODICALLY}</DescriptionLine>;
+};
+
+interface StepActionsRuleProps {
+  ruleActionsData: ActionsStepRule;
+  isRuleLoading: boolean;
+}
+
+export const StepActionsRule: React.FC<StepActionsRuleProps> = ({
+  ruleActionsData,
+  isRuleLoading,
+}) => {
+  const {
+    services: { triggersActionsUi },
+  } = useKibana();
+
+  const actionTypeRegistry = triggersActionsUi.actionTypeRegistry as ActionTypeRegistryContract;
+
+  const { data: connectors } = useFetchConnectors();
+  const { data: connectorTypes } = useFetchConnectorTypes();
+
+  const notificationActions = ruleActionsData.actions;
+  const responseActions = ruleActionsData.responseActions || [];
+
+  const ruleHasNoActions = notificationActions.length === 0 && responseActions.length === 0;
+
+  if (ruleHasNoActions || !connectors || !connectorTypes || !ruleActionsData) {
+    return null;
+  }
+
+  const hasBothNotificationAndResponseActions =
+    notificationActions.length > 0 && responseActions.length > 0;
+
+  return (
+    <EuiFlexItem data-test-subj="actions" component="section" grow={1}>
+      <StepPanel loading={isRuleLoading} title={i18n.ACTIONS}>
+        {hasBothNotificationAndResponseActions && (
+          <EuiText size="m">{i18n.NOTIFICATION_ACTIONS}</EuiText>
+        )}
+        <EuiSpacer size="s" />
+        {notificationActions.map((action) => (
+          <NotificationAction
+            action={action}
+            connectorTypes={connectorTypes}
+            connectors={connectors}
+            actionTypeRegistry={actionTypeRegistry}
+            key={action.id}
+          />
+        ))}
+        <EuiSpacer size="m" />
+
+        {hasBothNotificationAndResponseActions && (
```

Comment:
I guess the design was given to you like this, but I don't see why the `Response actions` and `Notification actions` titles should be displayed only when both are present. I think is still useful to know which type of actions you have in your rule, because they behave differently according to their type.